### ISSUE TITLE
let aliens evolve in their base even if outnumbered

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/gameplay#65]
Seen on @sweet235, alienPvE1, sol.

Previous title, when tagged as T-Bug:
aliens are unable to evolve near overmind when in sight of enemies

**Comment by Viech** [comment]
Is it a problem if aliens run at you with a dretch and then evolve right into your face though? Should we keep a minimum distance to enemies even if within your own base? Or some cooldown after attacking/being hurt?

**Comment by bmorel** [comment]
> Is it a problem if aliens run at you with a dretch and then evolve right into your face though? 

outside of A base, yes, but in A base, it's their only way to defend.

> Should we keep a minimum distance to enemies even if within your own base? 

Why not.

> Or some cooldown after attacking/being hurt?

Meh.

We could also only let dretches do that (evolve in sight) and only if distance is, i.e. 2 times the attack range of their desired class?
Or, we could simply immobilise aliens for, say, 1s while they evolve?

All those are suggestions, and values should probably be cvars so that we can more easily find a balance.

**Comment by Gireen** [comment]
I think the idea is pointless.
If there is a base then there are enough buildings to outnumber attackers.
If not then its no more a base and even then aliens can still evolve if they get out of sight of humans.

**Comment by Viech** [comment]
So what about _standing on creep_ as a condition? This is at least somewhat communicated through the health regeneration indicators.

**Comment by bmorel** [comment]
> If there is a base then there are enough buildings to outnumber attackers.

Depends. In PvP, yes, it's "very likely". In PvE? Not at all.

**Comment by ghost** [comment]
> So what about standing on creep as a condition? This is at least somewhat communicated through the health regeneration indicators.

Ah, so that creep would actually serve a purpose. Why not. But it needs to be increased, and in the current situation, it's possible to have creep traverse walls.

**Comment by mjstahlberg** [comment]
> > So what about standing on creep as a condition? This is at least somewhat communicated through the health regeneration indicators.
> 
> Ah, so that creep would actually serve a purpose. Why not. But it needs to be increased, and in the current situation, it's possible to have creep traverse walls.

One could additionally require a line of sight to the creep-producing buildable. Though I think it's better not to do so, because the odd case of being able to evolve at a wall next to the base is not as problematic as not being able to evolve within the base and visually on creep due to map geometry blocking the trace.

**Comment by ghost** [comment]
I agree with that, I just wanted to remind of that problem in case it slipped out of mind.

**Comment by mjstahlberg** [comment]
@Gireen would you be OK with tying enemy-observed evolution to *in creep range* instead of *number of friendly nearby buildables exceeds number of nearby enemy players*?

**Comment by Gireen** [comment]
I think that would make it even worse.
"in creep range" means nothing else then very close to a building. So if a single human would enter the base aliens could not evolve until they get close to a building. In a base with all buildings on walls and ceilings that leaves only the Overmind, which makes every dretch a easy target and defending impossible.

**Comment by DolceTriade** [comment]
I actually disagree. Being in creep range in your own base is kinda expected. The only thing I might be worried about is abusing this and being able to evolve at a forward base with only a booster or something.

I'd say evo near an egg or om regardless of situation. No more random alien evolution!

**Comment by mjstahlberg** [comment]
I was about to agree with @Gireen on the single human argument. :grin: 

Maybe tying it to a strict majority of friendlies over foes is not so bad … if communicated properly. And I don't know how to do that.

> The only thing I might be worried about is abusing this and being able to evolve at a forward base with only a booster or something. […] I'd say evo near an egg or om regardless of situation.

Your forward almost always has an egg (and if it doesn't then you're already evolved).

**Comment by Gireen** [comment]
> Being in creep range in your own base is kinda expected

Maybe i have the wrong definition of creep range but thats not true. As i understand it one is in creep range if there is a big second cross displayed, not like the small one around the overmind. 
If this is correct then it would exclude most of a base, and when buildings are moved to walls and ceiling it would again be impossible to evolve.

If the range is increased we get more room for abuse and sooner or later we get the current system or something similar

> Maybe tying it to a strict majority of friendlies over foes is not so bad

a group of Dretches turning to tyrants in front of you mid combat 😄

**Comment by mjstahlberg** [comment]
> > Maybe tying it to a strict majority of friendlies over foes is not so bad
> 
> a group of Dretches turning to tyrants in front of you mid combat smile

Right, I was thinking of the current approach but worded it badly. So strict majority of friendly buildables over enemy players.

**Comment by ghost** [comment]
To me, creep is the creepy texture that's emitted from buildings. This is also what is described as creep along the code (creep_slow in C++ for humans who would walk on it, creep size in config files).

I've never understood what was the point of the mysterious healing bonus aliens sometimes have, sometimes don't and to me, it could easily be removed as it feels... well, mysterious and a bit of deus ex machina. (I do know it's about radius from from something though, maybe overmind? Maybe any building building? But something. And unlike many players, I have read the code more than once, even if not for that particular information)

**Comment by ghost** [comment]
>  Though I think it's better not to do so, because the odd case of being able to evolve at a wall next to the base is not as problematic as not being able to evolve within the base and visually on creep due to map geometry blocking the trace.

Plus humans can buy from immune armories hidden behind glass, not to mention current creep and booster heal go through wall, so considering it a game mechanism (to not abuse) could be ok.

> a group of Dretches turning to tyrants in front of you mid combat smile

This is exactly what can happen in human bases if players use keybinds. Actually no, it's worst: it's luci suits with grenades, pulse rifle or lucis

**Comment by ghost** [comment]
so far it seems @Viech 's proposal: number of building over number of enemies, could work, but it does not at all. If you have not tried it yet, try `sol alienpve1` one of the most played alien training layouts. you'll see you're sometimes unable to evolve near overmind itself.

Maybe the solution is instant evol in LoS of overmind (it starts at us after all!) and to add some more constraint like only evolve out of sight if not protecting main base?
It would balance a bit the problem or armoury needs (but neither seen nor touchable, just about sheer distance which generates own lot of problems for bots... might have an idea, OT I know, but in case I forgot: mark the triangles in reach or armoury/medistat as beeing valid for such actions, this would allow to support several hacks used by mappers a clean way and to fix an ugly hack into hour codebase (that I'm the author of and only mitigates the problem)) for humans.

**Comment by Gireen** [comment]
i took a look at this `sol alienpve1` layout.
its an short corridor with overmind and a few buildings.
surrounded on both sides directly with human bases containing countless buildings

its a sad joke that so much time was wasted here to have a serious discussion about a gameplay change that would only serve an specific unrealistic scenario, while making the game worse everywhere else.

anyone with that PVE layout can set `g_evolveAroundHumans` to 0 for unrestricted evolving

in normal games there is no problem with the current system

**Comment by ghost** [comment]
> its a sad joke that so much time was wasted here to have a serious discussion

Well, in this discussions I learned that in fact, it's not about being close to A base, but about being outnumbered. I still think this is bad, as it prevents defending vs a rush, plus this could be used to besiege a base by spamming drills around it.

Also, how could I guess that info? If a dev contributor can't guess, how are players supposed to? We're always talking about simplifying the system: tell me how this is simple when you have no feedback you're outnumbered?

Last but not least:

> would only serve an specific unrealistic scenario

Not unrealistic since it will happens often in alien PVE and I do intend to build more. For your information, sol/alienpve1 is a very popular layout.

> while making the game worse everywhere else

Still has to be proven.

> anyone with that PVE layout can set g_evolveAroundHumans to 0 for unrestricted evolving

So, 1) thanks for information (again, no easy way to guess about those stuff). 2) that does not solves the problem which is specifically about "in their base".
